client/my-sites/email/email-management/home/email-plan-mailboxes-list.jsx

```diff
@@ -13,42 +15,73 @@ import SectionHeader from 'calypso/components/section-header';
 import Badge from 'calypso/components/badge';
 import { isEmailUserAdmin } from 'calypso/lib/emails';
 
-class EmailPlanMailboxesList extends React.Component {
-	render() {
-		const { emails, translate } = this.props;
+const MailboxListHeader = ( { children } ) => {
+	const translate = useTranslate();
 
-		if ( ! emails || emails.length < 1 ) {
-			return null;
-		}
+	return (
+		<div className="email-plan-mailboxes-list__mailbox-list">
+			<SectionHeader label={ translate( 'Mailboxes' ) } />
+			{ children }
+		</div>
+	);
+};
 
-		const emailsItems = emails.map( ( email ) => {
-			return (
-				<CompactCard
-					key={ `email-row-${ email.mailbox }` }
-					className="email-plan-mailboxes-list__mailbox-list-item"
-				>
+const MailboxListItem = ( { children, isPlaceholder, hasNoEmails } ) => {
+	const className = classNames( 'email-plan-mailboxes-list__mailbox-list-item', {
+		'is-placeholder': isPlaceholder,
+		'no-emails': hasNoEmails,
+	} );
+	return <CompactCard className={ className }>{ children }</CompactCard>;
+};
+
+function EmailPlanMailboxesList( { emails, isLoadingEmails } ) {
+	const translate = useTranslate();
+
+	if ( isLoadingEmails ) {
+		return (
+			<MailboxListHeader>
+				<MailboxListItem isPlaceholder>
 					<MaterialIcon icon="email" />
-					<span>
-						{ email.mailbox }@{ email.domain }
-					</span>
-					{ isEmailUserAdmin( email ) && (
-						<Badge type="info">
-							{ translate( 'Admin', {
-								comment: 'Email user role displayed as a badge',
-							} ) }
-						</Badge>
-					) }
-				</CompactCard>
-			);
-		} );
+					<span />
+				</MailboxListItem>
+			</MailboxListHeader>
+		);
+	}
 
+	if ( ! emails || emails.length < 1 ) {
 		return (
-			<div className="email-plan-mailboxes-list__mailbox-list">
-				<SectionHeader label={ translate( 'Mailboxes' ) } />
-				{ emailsItems }
-			</div>
+			<MailboxListHeader>
+				<MailboxListItem hasNoEmails>
+					<span>{ translate( 'No mailboxes' ) }</span>
```

Comment:
:warning: This change will be queued for retranslation. We'll display the English text until then.